api/payments.go

```diff
@@ -148,25 +148,22 @@ func (a *API) PaymentCreate(w http.ResponseWriter, r *http.Request) error {
 		return internalServerError("We failed to authorize the amount for this order: %v", err)
 	}
 	tr := models.NewTransaction(order)
-
-	order.PaymentProcessor = provider.Name()
 	processorID, err := charge(params.Amount, params.Currency)
 	tr.ProcessorID = processorID
 
 	if err != nil {
 		tr.FailureCode = strconv.FormatInt(http.StatusInternalServerError, 10)
 		tr.FailureDescription = err.Error()
-		tr.Status = "failed"
-	} else {
-		tr.Status = "pending"
```

Comment:
Everything I read on Stripe and PayPal's documentation said that if creating the charge is successful, it is paid.

If we were creating an "authorization", it would be a different story. Those you have to execute within a certain time frame (usually a week).